CONTRIBUTING.md

```diff
@@ -366,3 +366,74 @@ duration = float_or_none(video.get('durationMs'), scale=1000)
 view_count = int_or_none(video.get('views'))
 ```
 
+### Inline values
+
+Extracting variables is acceptable for reducing code duplication and improving readability of complex expressions. However, you should avoid extracting variables used only once and moving them to opposite parts of the extractor file, which makes reading the linear flow difficult.
+
+#### Example
+
+Correct:
+
+```python
+return self.playlist_result(
+    [self._parse_brightcove_metadata(vid, vid.get('id'), headers)
+        for vid in json_data.get('videos', []) if vid.get('id')],
+    json_data.get('id'), json_data.get('name'),
+    json_data.get('description'))
```

Comment:
As already pointed out, there should be more simple example with minimal unrelated code. Like:

Correct:
```python
title = self._search_regex(
    r'<span[^>]+class="title"[^>]*>([^<]+)', webpage, 'title')
```

Incorrect:
```python
_TITLE_RE = r'<span[^>]+class="title"[^>]*>([^<]+)'
... some lines of code ...
title = self._search_regex(_TITLE_RE, webpage, 'title')
```

Reply:
Okay, thanks. I simplified the example.